lib/stm32/f0/adc.c

```diff
@@ -241,7 +241,7 @@ bool adc_get_watchdog_flag(uint32_t adc)
 
 void adc_clear_watchdog_flag(uint32_t adc)
 {
-	ADC_ISR(adc) = ADC_ISR_AWD1;
+	ADC_ISR(adc) |= ADC_ISR_AWD1;
```

Comment:
According to F0 Reference manual, bits in ADC_ISR register are cleared by writing 1 to them.
By or-ing ADC_ISR with AWD1 bit your code will actually clear all set ISR bits (and awd1) in ISR register, which is not what you want.

Reply:
:eyes:  I'll try to deal with this revelation and update it this weekend.

Reply:
i ditched the commit.